File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

letters = 0

words = 0

sentences = 0

for letter in text:
    if ((letter >= 'A' and letter <= 'Z') or (letter >= 'a' and letter <= 'z')):
        letters += 1

    if (letter == ' '):
        words += 1

    if (letter == '.' or letter == '!' or letter == '?'):
        sentences += 1

if (text[len(text) - 1] != ' '):
    words += 1

lettersPer100Words = letters / words * 100

sentencesPer100Words = sentences / words * 100

index = 0.0588 * lettersPer100Words - 0.296 * sentencesPer100Words - 15.8

if index < 1:
    print("Before Grade 1")
elif (index >= 16):
    print("Grade 16+")
else:
    print(f"Grade {round(index)}")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string text = "";

    text = get_string("Text: ");

    int letters = 0;

    int words = 0;

    int sentences = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if ((text[i] >= 'A' && text[i] <= 'Z') || (text[i] >= 'a' && text[i] <= 'z'))
        {
            letters++;
        }

        if (text[i] == ' ')
        {
            words++;
        }

        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    if (text[strlen(text) - 1] != ' ')
    {
        words++;
    }

    float lettersPer100Words = (float)letters / (float)words * 100;

    float sentencesPer100Words = (float)sentences / (float)words * 100;

    float index = 0.0588 * lettersPer100Words - 0.296 * sentencesPer100Words - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", (int)round(index));
    }
}